nncf/pruning/filter_pruning/algo.py

```diff
@@ -491,12 +508,13 @@ def _set_binary_masks_for_pruned_modules_globally_by_flops_target(self,
         cur_num = 0
         tmp_in_channels = self.modules_in_channels.copy()
         tmp_out_channels = self.modules_out_channels.copy()
+        tmp_pruning_quotas = self.pruning_quotas.copy()
         while cur_num < len(sorted_importances):
             cluster_idx = int(sorted_importances[cur_num][1])
             filter_idx = int(sorted_importances[cur_num][2])
 
-            if self.pruning_quotas[cluster_idx] > 0:
-                self.pruning_quotas[cluster_idx] -= 1
+            if tmp_pruning_quotas[cluster_idx] > 0:
```

Comment:
Consider rounding the pruning quotas down to the nearest integer on initialization so that we don't accidentally prune more filters than we want.

Reply:
Done.